modules/tinymce/src/themes/silver/test/ts/webdriver/editor/DisabledMenuItemTest.ts

```diff
@@ -0,0 +1,53 @@
+import { Keys, Mouse, RealKeys, UiFinder, Waiter } from '@ephox/agar';
+import { afterEach, beforeEach, describe, it } from '@ephox/bedrock-client';
+import { TinyHooks, TinyUiActions } from '@ephox/mcagar';
+import { SugarBody } from '@ephox/sugar';
+
+import { Editor } from 'tinymce/core/api/PublicApi';
+import TablePlugin from 'tinymce/plugins/table/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('webdriver.tinymce.themes.silver.editor.menubar.DisabledMenuItemTest', () => {
+
+  const verticalAlignMenuItemSelector = '[title="Vertical align"]';
+  const tableMenuItemSelector = '[role="menuitem"]:contains("Table")';
+
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    plugins: 'table',
+    menu: {
+      table: { title: 'Table', items: 'inserttable | cell row column | tableclass tablecellclass tablecellvalign tablecellborderwidth tablecellborderstyle tablecaption tablecellbackgroundcolor tablecellbordercolor | advtablesort | tableprops deletetable' }
+    },
+    menubar: 'table',
+  }, [ Theme, TablePlugin ]);
+
+  const pOpenTableMenu = () => {
+    const editor = hook.editor();
+    TinyUiActions.keydown(editor, Keys.escape()); // to close table menu before opening
+    TinyUiActions.clickOnMenu(editor, tableMenuItemSelector);
+    return Waiter.pTryUntil('Wait for table menu to open', () => UiFinder.exists(SugarBody.body(), verticalAlignMenuItemSelector));
+  };
+
+  const assertVerticalAlignMenuIsNotOpen = () => {
+    UiFinder.notExists(SugarBody.body(), '[role="menuitemcheckbox"]:contains("None")');
+    UiFinder.notExists(SugarBody.body(), '[role="menuitemcheckbox"]:contains("Top")');
+    UiFinder.notExists(SugarBody.body(), '[role="menuitemcheckbox"]:contains("Middle")');
+    UiFinder.notExists(SugarBody.body(), '[role="menuitemcheckbox"]:contains("Bottom")');
+  };
+
+  beforeEach(async () => {
+    await pOpenTableMenu();
+  });
+
+  afterEach(() => {
+    assertVerticalAlignMenuIsNotOpen();
```

Comment:
Assertions should never be done in hooks, they should only contain setup and teardown logic.

- [ ] Revert this change please